IPA enable pcm for sle, fix parallel run of ipa.pm

cfconrad opened the pull request:
Use a stable image to run the IPA tool. If the distri is sle, we enable
the pcm module manually. 

Upload JSON results file with parse_extra_log().

FIX: When running ipa.pm in parallel with the same AWS account, the cleanup also terminated the instances of the other jobs. Now we do only terminate the instance which was actually created.
Also the SSH key name was made unique per job.

- Verification run:http://fromm.arch.suse.de/tests/981

okurz commented on tests/publiccloud/ipa.pm:
```diff
@@ -41,21 +42,44 @@ sub create_secgroup {
+    for my $i (0 .. 9) {
+        my $key_name = $prefix . "_" . $i;
+        my $cmd      = "aws ec2 create-key-pair --key-name '" . $key_name . "' --query 'KeyMaterial' --output text > " . $out_file;
+        if (script_run($cmd) == 0) {
```
Why not just use `assert_script_run` when you would die in L55 anyway?

cfconrad replied:
I use a different name on the next try, so It could work. And die only after ten retries.

okurz commented on tests/publiccloud/ipa.pm:
```diff
@@ -41,21 +42,44 @@ sub create_secgroup {
 
     die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
 
+    if (check_var('DISTRI', 'sle')) {
```
Please use `is_sle` from lib/version_utils.pm

cfconrad replied:
ok

okurz commented on tests/publiccloud/ipa.pm:
```diff
@@ -41,21 +42,44 @@ sub create_secgroup {
     die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
 
+    if (check_var('DISTRI', 'sle')) {
+        my $modver = get_required_var("VERSION") =~ s/-SP\d+//gr;
```
Please use single-ticks by default unless you want to evaluate variables within strings. I wonder why we need this special treatment of the version here. If this is something that is necessary in more places we can also move it into the `add_suseconnect_product` call

cfconrad replied:
The module version of public-cloud doesn't contain the service-pack number. I don't see a reason why some modules have a SP number and others not.

cfconrad replied:
Thanks for pointing to that link.

okurz commented on tests/publiccloud/ipa.pm:
```diff
@@ -104,28 +129,29 @@ sub run {
-        }
+    # Terminate instance
+    if (defined($self->{'ipa_instance_id'})) {
+        script_run("aws ec2 terminate-instances --instance-ids " . $self->{'ipa_instance_id'});
```
Why would you like to ignore failures in this call? If the termination fails, isn't that something critical we want to catch? Or could it be that the instance already terminated in a different way in before? I would be more explicit about this and in general recommend to use `script_run` only when you use the return code

cfconrad replied:
right, I will change that.

okurz commented:
Can you comment why the failing job run should count as a "verification run" still?

cfconrad commented:
True, a passed job is better for verification run. Why the test_sles_repos_py__test_sles_repos fail needs further investigation, I will take a look.
I added a variable to pass the IPA test names and removed the tests_sles_on_demand tests for now.
http://fromm.arch.suse.de/tests/995
